### ISSUE TITLE
ipv8 upgrades to: based SHA-3 decentralized IP address, ipv4 ipv6 terminator

### ISSUE THREAD

**Issue by Pantyhose-X** [Tribler/py-ipv8#1247]
Paper should be written and published in IEEE
**Title:** based SHA-3 decentralized IP address, ipv4 ipv6 terminator

* [SHA-3](https://en.wikipedia.org/wiki/SHA-3) address + decentralized DNS, access to decentralized domain

win10 network adapters : Enter the [SHA-3](https://en.wikipedia.org/wiki/SHA-3) address
Wi-Fi  DHCP Static : Enter the [SHA-3](https://en.wikipedia.org/wiki/SHA-3) address

---
IPv4 and IPv6 are centralized. They are controlled by national companies, etc. 
![](https://www.apnic.net/wp-content/uploads/policy-environment_obsolete/images/ipv4_hierarchy.png)

Try to Google “Who controls IP addresses?” You’ll promptly get “IANA: the Internet Assigned Numbers Authority.” IANA is the top authority behind IP address allocation and assignment. There are five different regional internet registries (RIR) with jurisdiction under the IANA.

As a matter of fact, as an individual or a normal internet user you cannot request IP addresses directly from IANA or one of the five RIRs, but only from internet service providers, such as the services offered by mobile or telecom operators.

* True decentralization, no official singular Foundation, Committee, Corporation, or entities in permanent unitary control of the protocol.

**Comment by qstokkink** [comment]
Thank you for your suggestion. It would certainly be nice to do away with IPv4 and IPv6 and - in general - centralized entities on the Internet.

Regarding your suggestion, our [Kotlin IPv8 implementation](https://github.com/Tribler/kotlin-ipv8) can create P2P connections using Bluetooth. If you're looking for IEEE papers in particular, you can look into WiFi standard IEEE 802.11.

SHA-3 is simply a hashing standard and will not - by itself - help with superseding IP addresses.

All of that said, py-IPv8 operates on the application layer of your operating system. For now, at least, our project scope does not include modifications to the network layer. The goal of our project is to provide a framework for P2P anonymisation for average users without needing elevated permissions on your operating system or requiring our users to buy custom hardware.

In summary, even though we believe in your ideals, we cannot make them a part of this project and I'll close this issue.
